client/client.go

```diff
@@ -488,6 +490,73 @@ func (c *client) GetRegion(ctx context.Context, key []byte) (*Region, error) {
 	return c.parseRegionResponse(resp), nil
 }
 
+func (c *client) waitForRegion(ctx context.Context, key []byte) (*pdpb.GetRegionResponse, error) {
+	childContext, cancel := context.WithCancel(ctx)
+
+	ch := make(chan *pdpb.GetRegionResponse, len(c.connMu.clientConns))
+
+	c.connMu.RLock()
+	for _, conn := range c.connMu.clientConns {
+		go c.getRegionResponse(childContext, pdpb.NewPDClient(conn), key, ch)
```

Comment:
we can try it one by one, do not request all PDs, which will make the request multiply increase.